docs/test-framework-dev-guide.md

```diff
@@ -12,14 +12,49 @@ Go version should be at least the same than the one in [.go-version](https://git
 
 ESS (QA) API Key to create on https://console.qa.cld.elstc.co/deployment-features/keys
 
-Warning: if you never created a deployment on it, you won't have permission to get this key so you will need to create one first.
+Warning: if you never created a deployment on it, you won't have permission to get this key, so you will need to create one first.
+
+## Running tests
+
+Some integration and E2E tests are safe to run locally. These tests set
+`Local: true` in their test functions' `define.Require` directive. Tests that
+don't set `Local: true` or explicitly set `Local: false` are not considered
+safe to run locally and will be executed on remote VMs instead.
+
+The framework will look for the agent version defined by the `AGENT_VERSION`
+environment variable, even for local tests, **regardless of what was defined in
+the test Fixture**. If `AGENT_VERSION` isn't set, it'll default to the current
+version without SNAPSHOT.
+
+### Setup
+One-time setup is required to run any integration and E2E tests. Run
+`mage integration:auth` to perform this setup.
+
+### Running the tests
+
+The test are run with mage using the `integration` namespace:
+
+- `mage integration:test` to execute all tests under the `testing/integration`
+  folder. All tests are executed on remote VMs, including those that set `Local: true`.
+
+- `mage integration:local [testName|all]` to execute only those tests under the
+  `testing/integration` folder that set `Local: true`. It'll run all the tests if
+`all` is passed as argument, or it'll pass `[testName]` to `go test` as
+- `--run=[testName]`.These tests are executed on your local machine.
+
+- `mage integration:local [testName]` same as `mage integration:local`, but it'll
```

Comment:
Just noticed on re-read, this is documenting `mage integration:local` a second time.